go/cmd/dolt/commands/sql.go

```diff
@@ -927,36 +945,49 @@ const (
 )
 
 type sqlEngine struct {
-	sdb          *dsqle.Database
+	sdb          dsqle.Database
 	ddb          *doltdb.DoltDB
 	engine       *sqle.Engine
 	resultFormat resultFormat
 }
 
 // sqlEngine packages up the context necessary to run sql queries against sqle.
-func newSqlEngine(ctx context.Context, dEnv *env.DoltEnv, db *dsqle.Database, format resultFormat) (*sqlEngine, error) {
+func newSqlEngine(ctx context.Context, dEnv *env.DoltEnv, root *doltdb.RootValue, db dsqle.Database, format resultFormat) (*sqlEngine, *sql.Context, error) {
```

Comment:
Similar to comments on your other PR, I don't like this coupling (returning a sql engine and the context required to use it). Would prefer a method on the sqlEngine object that creates new context objects on demand.

Reply:
This is my main comment on this PR. The rest of it looks fine.

Reply:
uncoupled.  creating the sql.Context externally and passing it in.